src/tracker-component.js

```diff
@@ -254,6 +271,35 @@ const clearExpiredCart = () => {
     }
 };
 
+const getPropertyId = ({ paramsToPropertyIdUrl }) => {
+    return new Promise(resolve => {
+        const clientId = getClientID();
+        const merchantId = getMerchantID()[0];
+        const propertyIdKey = `property-id-${ clientId }-${ merchantId }`;
+        const savedPropertyId = window.localStorage.getItem(propertyIdKey);
+        const currentUrl = `${ window.location.protocol }//${ window.location.host }`;
+        if (savedPropertyId) {
+            return resolve(savedPropertyId);
+        }
+        let url;
+        if (paramsToPropertyIdUrl) {
+            url = paramsToPropertyIdUrl();
+        } else {
+            url = 'https://paypal.com/tagmanager/containers/xo';
+        }
+        fetch(`${ url }?mrid=${ merchantId }&url=${ encodeURIComponent(currentUrl) }`)
+            .then(res => {
+                if (res.status === 200) {
+                    return res;
+                }
```

Comment:
Should we handle the case here if `res.status` isn't 200?

Reply:
could we TD this to add proper logging?